core/src/main/java/com/amplifyframework/core/model/AuthRule.java

```diff
@@ -54,6 +55,9 @@
      */
     public AuthRule(com.amplifyframework.core.model.annotations.AuthRule authRule) {
         this.authStrategy = authRule.allow();
+        this.authProvider = Empty.check(authRule.provider()) ?
+                                        authStrategy.getDefaultAuthProvider() :
+                                        AuthStrategy.Provider.valueOf(authRule.provider());
```

Comment:
looks like you already do some null checking in the getter later so maybe this should just be `authRule.provider()`?

Reply:
Actually, I have to do something at this level because I need to translate from a String to an enum item. I'm going to tweak this so the null check only happens in one place though.